### PR TITLE
net/mwan3: fixes for mwan3 (lede-17.01)

### PR DESCRIPTION
**Pull request by feckert**
Maintainer: me

Compile tested: n/a (only changed a shell script)
Run tested: (lantiq_xrx200, LEDE 17.01.0, tested failover and failback; iptables/ipset/ip checked)

Description:
Add some fixes to mwan3 cherry picked from master (with minor changes to get commits applied)
Tested-by: @rmilecki

### CODE REVIEW

File: net/mwan3/Makefile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #
-# Copyright (C) 2006-2014 OpenWrt.org
+# Copyright (C) 2006-2017 OpenWrt.org
```
Comment (champtar) [inline net/mwan3/Makefile]: As you are not under contract with OpenWrt, you should not update the OpenWrt copyright

File: net/mwan3/Makefile [hunk 2]
```diff
@@ -8,9 +8,9 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=mwan3
-PKG_VERSION:=2.0
-PKG_RELEASE:=3
-PKG_MAINTAINER:=Jeroen Louwes <jeroen.louwes@gmail.com>
+PKG_VERSION:=2.0.1
+PKG_RELEASE:=4
```
Comment (champtar) [inline net/mwan3/Makefile]: when you increase PKG_VERSION, in general you reset PKG_RELEASE to 1
I haven't checked but if you backported all patches from master, maybe use the same version

File: net/mwan3/Makefile [hunk 3]
```diff
@@ -21,7 +21,7 @@ define Package/mwan3
    SUBMENU:=Routing and Redirection
    DEPENDS:=+ip +ipset +iptables +iptables-mod-conntrack-extra +iptables-mod-ipopt
    TITLE:=Multiwan hotplug script with connection tracking support
-   MAINTAINER:=Jeroen Louwes <jeroen.louwes@gmail.com>
+   MAINTAINER:=Florian Eckert <fe@dev.tdt.de>
```
Comment (champtar) [inline net/mwan3/Makefile]: you don't need MAINTAINER, the global PKG_MAINTAINER is enough

File: net/mwan3/Makefile (hunk 1 above)
Comment (feckert) [inline net/mwan3/Makefile]: @champtar ok i will drop this.
But some did some did not change this!
When should i update this?
My information is if i change the Makefile i have to update the year.

File: net/mwan3/Makefile (hunk 2 above)
Comment (feckert) [inline net/mwan3/Makefile]: @champtar I know but the source are in your repo so the PKG_RELEASE would always be set to '1' the only exemption would be if i update the make file and not the sources.
I have talked to @rmilecki and the commits are bugfixes scattered over different version in the master. He agreed that i should use a version which only fits to this branch.

File: net/mwan3/Makefile (hunk 3 above)
Comment (feckert) [inline net/mwan3/Makefile]: @champtar on master i have updated this already. So i thought I should change this in this branch as well when i add some changes. @Adze1502 dropped the support for this package in master already. I agreed to maintenance this package for him.

**Comment by rmilecki** [comment]
On 21 August 2017 at 14:56, Florian Eckert <notifications@github.com> wrote:
> > @@ -1,5 +1,5 @@
>  #
> -# Copyright (C) 2006-2014 OpenWrt.org
> +# Copyright (C) 2006-2017 OpenWrt.org
>
> @champtar ok i will drop this.
> But some did some did not change this!
> When should i update this?
> My information is if i change the Makefile i have to update the year.

People shouldn't update it or assign copyrights to OpenWrt in the first place.

If you do something copyrightable you can add something like:
Copyright (C) 2017 Florian <florian@...>

**Comment by feckert** [comment]
@rmilecki thanks for clarify i will leave it as it is in the future thanks

File: net/mwan3/Makefile (hunk 3 above)
Comment (champtar) [inline net/mwan3/Makefile]: My point was update PKG_MAINTAINER line, remove MAINTAINER line

File: net/mwan3/Makefile (hunk 2 above)
Comment (champtar) [inline net/mwan3/Makefile]: agreed, if code is different, version MUST be different so it's easier for bug reports

**Comment by feckert** [comment]
@champtar fixed outstanding changes in Makefile except version string

**Comment by rmilecki** [comment]
On 21 August 2017 at 14:57, Florian Eckert <notifications@github.com> wrote:
> @champtar I know but the source are in your repo so the PKG_RELEASE would always be set to '1' the only exemption would be if i update the make file and not the sources.
> I have talked to @rmilecki and the commits are bugfixes scattered over different version in the master. He agreed that i should use a version which only fits to this branch.

It's indeed quite a specific package as its whole sources are in the
package itself. After thinking about this it probably doesn't make
sense to create version 2.0.1 and start counting its releases from 4.
So I'd suggest picking *one* of the following updates:
1) 2.0-3 to 2.0-4
or
2) 2.0-3 to 2.0.1-1

I don't have any strong preference for any of above. Keeping
PKG_RELEASE at 1 unless only the Makefile gets changed may be a good
idea if you prefer it this way.

**Comment by feckert** [comment]
@rmilecki fixed it. Pick the version 2.0.1-1

**Pull request merged**